cpp/src/arrow/dataset/dataset.cc

```diff
@@ -16,13 +16,15 @@
 // under the License.
 
 #include "arrow/dataset/dataset.h"
+#include <arrow/util/thread_pool.h>
```

Comment:
Let's be careful to use the same inclusion style - quotes for Arrow includes, brackets for other ones.

Reply:
Sure, thanks for noting.

Reply:
I'll add some basic tests in the next iteration.